test/lib/factories/PathStartFactory.test.ts

```diff
@@ -0,0 +1,194 @@
+
+	beforeEach(() => {
+		process.chdir(PATH_TO_WORKSPACE);
+	});
+
+	afterEach(() => {
+		process.chdir(ORIGINAL_TEST_DIR);
+	})
+
+	it('When passed undefined, returns undefined', async () => {
+		const factory = new PathStartFactory();
+		const output = await factory.create(undefined);
+		expect(output).toBeUndefined();
+	});
+
+	it('When passed an empty array, returns an empty array', async () => {
+		const factory = new PathStartFactory();
+		const output = await factory.create([]);
+		expect(output).toEqual([]);
+	});
+
+	describe('File matching', () => {
+		it.each([
+			{scenario: 'a single relative file', expectation: 'just that file', input: ['undotted-file-1.txt']},
+			{scenario: 'a list of relative files', expectation: 'just those files', input: [path.join('.', 'undotted-directory-a', 'undotted-file-2a.txt'), '.dotted-file-2.txt']},
+			{
+				scenario: 'a single absolute file',
+				expectation: 'just that file',
+				input: [path.join(PATH_TO_WORKSPACE, 'undotted-file-1.txt')]
+			},
+			{
+				scenario: 'a list of absolute files',
+				expectation: 'just those files',
+				input: [
+					path.join(PATH_TO_WORKSPACE, 'undotted-directory-a', 'undotted-file-2a.txt'),
+					path.join(PATH_TO_WORKSPACE, 'undotted-file-2.txt')
+				]
+			}
+		])('Given $scenario, returns $expectation', async ({input}) => {
+			const factory = new PathStartFactory();
+			const output = await factory.create(input);
+			expect(output).toBeDefined();
+			expect(output as string[]).toHaveLength(input.length);
+			expect(output as string[]).toEqual(input);
+		})
+	});
+
+	describe('Glob matching', () => {
+		it.each([
+			{
+				scenario: 'a single relative file',
+				glob: path.join('.', 'undotted*1.txt'),
+				expectation: [
+					expect.stringMatching('undotted-file-1\\.txt')
+				]
+			},
+			{
+				scenario: 'a list of relative files',
+				glob: path.join('.', 'undotted*'),
+				expectation: [
+					expect.stringMatching('undotted-file-1\\.txt'),
+					expect.stringMatching('undotted-file-2\\.txt')
+				]
+			},
+			{
+				scenario: 'a single absolute file',
+				glob: path.resolve(PATH_TO_WORKSPACE, 'undotted*1.txt'),
+				expectation: [
+					expect.stringMatching('undotted-file-1\\.txt')
+				]
+			},
+			{
+				scenario: 'a list of absolute files',
+				glob: path.resolve(PATH_TO_WORKSPACE, 'undotted*'),
+				expectation: [
+					expect.stringMatching('undotted-file-1\\.txt'),
+					expect.stringMatching('undotted-file-2\\.txt')
+				]
+			}
+		])('Given a glob that matches $scenario, returns the correct files', async ({glob, expectation}) => {
+			const factory = new PathStartFactory();
+			const output = await factory.create([glob]);
+			expect(output).toBeDefined();
+			expect(output as string[]).toHaveLength(expectation.length);
+			expect(output as string[]).toEqual(expect.arrayContaining(expectation));
+		});
+
+		describe('Dotted file handling', () => {
```

Comment:
Since we just got bit by some stuff with this, I opted to add aggressive test coverage to lock in behavior.